source/lwip/iot_socket.c

```diff
@@ -611,15 +619,21 @@ int32_t iotSocketSetOpt (int32_t socket, int32_t opt_id, const void *opt_val, ui
       }
       break;
     case IOT_SOCKET_SO_RCVTIMEO:
-      rc = setsockopt(socket, SOL_SOCKET, SO_RCVTIMEO,  (const char *)opt_val, opt_len);
-      if (rc == 0) {
-        sock_attr[socket-LWIP_SOCKET_OFFSET].tv_sec  = *(const uint32_t *)opt_val / 1000U;
-        sock_attr[socket-LWIP_SOCKET_OFFSET].tv_msec = *(const uint32_t *)opt_val % 1000U;
+    case IOT_SOCKET_SO_SNDTIMEO: ; /* do not remove the semi-colon */
```

Comment:
I would add to the commit message the intention with `;` being placed here. It's important detail why are we doing this.